packages/superbridge/src/Superbridge.ts

```diff
@@ -21,11 +21,117 @@ import {
 import { Chains } from '@rabbitholegg/questdk-plugin-utils'
 import { type Address, zeroAddress } from 'viem'
 
+function getInputsFromEthereum(params: BridgeActionParams) {
+  const { tokenAddress, amount, recipient } = params
+  const ercInputs = [
+    {
+      $abi: [BRIDGE_ERC20_FRAGMENT],
+      _localToken: tokenAddress,
+      _amount: amount,
+    },
+    {
+      $abi: [BRIDGE_ERC20_TO_FRAGMENT],
+      _localToken: tokenAddress,
+      _to: recipient,
+      _amount: amount,
+    },
+    {
+      $abi: [DEPOSIT_ERC20_FRAGMENT],
+      _l1Token: tokenAddress,
+      _amount: amount,
+    },
+    {
+      $abi: [DEPOSIT_ERC20_TO_FRAGMENT],
+      _l1Token: tokenAddress,
+      _amount: amount,
+      _to: recipient,
+    },
+  ]
+  const ethInputs = [
+    {
+      $abi: [BRIDGE_ETH_FRAGMENT, DEPOSIT_ETH_FRAGMENT],
+    },
+    {
+      $abi: [BRIDGE_ETH_TO_FRAGMENT, DEPOSIT_ETH_TO_FRAGMENT],
```

Comment:
All these ABI functions were combined in an $or before. Now it splits them based on the `tokenAddress`.

Now these will only be accesses if tokenAddress is ETH, or tokenAddress is 'Any'